cmd/image-builder/main_test.go

```diff
@@ -949,3 +953,81 @@ func TestBasenameFor(t *testing.T) {
 		assert.Equal(t, tc.expected, main.BasenameFor(res, tc.basename))
 	}
 }
+
+// XXX: move into as manifestgen.FakeDepsolve
+func fakeDepsolve(cacheDir string, depsolveWarningsOutput io.Writer, packageSets map[string][]rpmmd.PackageSet, d distro.Distro, arch string) (map[string]dnfjson.DepsolveResult, error) {
+	depsolvedSets := make(map[string]dnfjson.DepsolveResult)
+
+	for name, pkgSetChain := range packageSets {
+		specSet := make([]rpmmd.PackageSpec, 0)
+		for _, pkgSet := range pkgSetChain {
+			include := pkgSet.Include
+			slices.Sort(include)
+			for _, pkgName := range include {
+				checksum := fmt.Sprintf("%x", sha256.Sum256([]byte(pkgName)))
+				spec := rpmmd.PackageSpec{
+					Name:     pkgName,
+					Checksum: "sha256:" + checksum,
+				}
+				specSet = append(specSet, spec)
+			}
+
+			depsolvedSets[name] = dnfjson.DepsolveResult{
+				Packages: specSet,
+				Repos:    pkgSet.Repositories,
+			}
+		}
+	}
+
+	return depsolvedSets, nil
+}
+
+func TestManifestIntegrationWithRegistrations(t *testing.T) {
+	restore := main.MockManifestgenDepsolver(fakeDepsolve)
+	defer restore()
+
+	restore = main.MockNewRepoRegistry(testrepos.New)
+	defer restore()
+
+	// XXX: only "proxy_123", "server_url_123" are actually observable
+	// in the manifest(?)
+	fakeRegContent := `{
+  "redhat": {
+    "subscription": {
+      "activation_key": "ak_123",
+      "organization": "org_123",
+      "server_url": "server_url_123",
+      "base_url": "base_url_123",
+      "insights": true,
+      "rhc": true,
+      "proxy": "proxy_123"
+    }
+  }
+}`
+	fakeRegistrationsPath := filepath.Join(t.TempDir(), "registrations.json")
+	err := os.WriteFile(fakeRegistrationsPath, []byte(fakeRegContent), 0644)
+	assert.NoError(t, err)
+
+	// XXX: fake the depsolving or we will need full support for
+	// subscripbed hosts in the tests
+	restore = main.MockOsArgs([]string{
+		"manifest",
+		"qcow2",
+		"--arch=x86_64",
+		"--distro=rhel-9.6",
+		"--registrations", fakeRegistrationsPath,
+	})
+	defer restore()
+
+	var fakeStdout bytes.Buffer
+	restore = main.MockOsStdout(&fakeStdout)
+	defer restore()
+
+	err = main.Run()
+	assert.NoError(t, err)
+
+	// XXX: manifesttest really needs to grow more helpers
```

Comment:
Just saying.... :-)

```
$ ag XXX | wc -l
78
```

Reply:
*cough* yeah :( sorry. I will open a PR with moving this into images right away

Reply:
I did not mean you to force to do anything, sorry. Maybe GH issues instead of `XXX` strings could be better but no biggie.